libcaliptra/test/hwmod_test/hwmod_if_impl.c

```diff
@@ -0,0 +1,82 @@
+//Licensed under the Apache-2.0 license
+
+#include <stdint.h>
+#include <stddef.h>
+#include <stdbool.h>
+
+#include "caliptra_model.h"
+#include "caliptra_api.h"
+
+#define CALIPTRA_STATUS_OK 0
+
+// Implementation specifics
+
+struct caliptra_model_init_params init_params;
+struct caliptra_fuses fuses = {0};
+
+static bool caliptra_model_init_complete = false;
+static struct caliptra_model *model = NULL;
+
+struct caliptra_model* hwmod_get_or_init(void)
+{
+    if (model == NULL)
+    {
+        int status = caliptra_model_init_default(init_params, &model);
+    }
+
+    return model;
+}
+
+// Memory
+
+/**
+ * caliptra_write_u32
+ *
+ * Writes a uint32_t value to the specified offset.
+ *
+ * @param[in] addr Memory address to write
+ * @param[in] data Data to write at offset
+ *
+ * @return 0 if successful, other if error (TBD)
+ */
+int caliptra_write_u32(uint32_t addr, uint32_t data)
+{
+    struct caliptra_model *m = hwmod_get_or_init();
+
+    caliptra_model_step(m);
+    int result = caliptra_model_apb_write_u32(m, addr, (int)data);
+    caliptra_model_step(m);
+
+    return result; 
+}
+
+/**
+ * caliptra_read_u32
+ *
+ * Writes a uint32_t value to the specified offset.
+ *
+ * @param[in] addr Memory address to read
+ * @param[in] data Pointer to a uint32_t to store the data
+ *
+ * @return 0 if successful, other if error (TBD)
+ */
+int caliptra_read_u32(uint32_t addr, uint32_t *data)
+{
+    struct caliptra_model *m = hwmod_get_or_init();
+
+    caliptra_model_step(m);
+    int result = caliptra_model_apb_read_u32(hwmod_get_or_init(), addr, (int*)data);
```

Comment:
Un-needed second call to hwmod_get_or_init

Reply:
Fixed